src/lib/scheduler/worker.cpp

```diff
@@ -17,6 +17,8 @@
 
 namespace {
 
+static const auto MAX_SLEEP_TIME_NS = uint32_t{10000000};  // 10 ms
```

Comment:
I think the goto modern fancy way is to make this constexpr auto. 

Also, would it be possible to make the type of this (and _sleep_time_ns) 
`std::chrono::nanoseconds`?